pyparsing/diagram/__init__.py

```diff
@@ -161,7 +218,7 @@ def __init__(
         #: The order in which we found this element, used for sorting diagrams if this is extracted into a diagram
         self.number = number  # type: int
         #: The index of this inside its parent
-        self.parent_index = index  # type: Optional[int]
+        self.parent_index = parent_index  # type: Optional[int]
```

Comment:
I guess since you've moved to Python 3.6 with type annotations, these can be pulled out as full annotations instead of type comments?

Reply:
Will do.